azavea/ckanext-odp_theme
# Upgrade odp_theme for CKAN 2.8

### Overview

Upgrades the overridden templates to work with CKAN 2.8 and override the 2.8 templates.

I tried to get upstream updates and apply our overrides by sometimes copying the new upstream template and applying the changes we had made to the old template and sometimes working from the existing overridden template to apply upstream changes.  I.e. "copy then modify" if the upstream changes were extensive and/or our changes were small, "apply updates manually" if the upstream changes were minor and/or our modifications were extensive (especially in cases where we're only overriding one block).

In a couple places, more blocks were added upstream so I could reduce the amount of unmodified code that we have copied into our templates.

~Large parts of this are working, but some parts are crashing.  Most frustratingly, the home view crashes in a way that, as far as I can tell, doesn't send the error to the logs.  I.e. something inside the `layout1.html` snippet is crashing but the error is getting swallowed.~ See below for update.

### Notes

Thanks to @designmatty for sitting down with me and working out the fixes to make the home page look right.

The most noticeable upstream changes to the styling are that the pages are wider and they use bootstrap column classes (rather than bootstrap-reminiscent classes like "span6").

### To test

- Check out this branch in your `ckanext-odp_theme` directory.
- Check out develop in your `opendataphilly-ckan` directory.
- In the `deployment/ansible/app.yml` file in `opendataphilly-ckan`, uncomment the `- { role: "ckanext-odp_theme" }` line.
- Provision according to the instructions in https://github.com/azavea/opendataphilly-ckan/pull/76, but with the added wrinkle that the database machine currently needs to be provisioned on commit [3a872c0a](https://github.com/azavea/opendataphilly-ckan/commit/3a872c0a4d7a73dc9375fdcb58f1f1e60b043ba9)
- Navigate the site.  It should look like the production site (see https://github.com/azavea/urban-apps/issues/147 for a workaround for the missing images) except for the width change.
- Crashes or layout changes that look bad are bugs.  Layout changes that look fine are upgrades :grin:

Resolves https://github.com/azavea/urban-apps/issues/158
